### PR TITLE
Add track events to customize store transitional page

### LINKED ISSUE #40128: Customize store Transitional page - Tracks
### Tracks

P2: pdnLyh-4d9-p2 (Mainly focus on high-priority tracks; feel free to add medium tracks if you're keen)

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_transitional_preview_store_click`   |
| What does the track measure      | % of users that click on “Preview store”.  |
| When does the track fire      | When a user clicks on “Preview store” button |
| Event properties     |  |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_transitional_editor_click`   |
| What does the track measure      | % of users that click on “Go to the Editor”. |
| When does the track fire      | When a user clicks on “Go to the Editor” button |
| Event properties     |  |

| Setting      | Value |
| ----------- | ----------- |
| Name      |  `wcadmin_customize_your_transitional_home_click`   |
| What does the track measure      | % of users that click on “Back to Home”. |
| When does the track fire      | When a user clicks on “Back to Home” button |
| Event properties     |  |

Note: After creating these tracks in code, you'll also need to PCYsg-dY7-p2 the tracks in order for them to be used in funnels and ExPlat.

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40128.

Adds tracks to monitor:

- % of users that click on “Preview store”.
- % of users that click on “Go to the Editor”.
- % of users that click on “Back to Home”.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Open your browser developer console and run `localStorage.setItem( 'debug', 'wc-admin:*' );` to set debug for tracks
5. Go to `WooCommerce > Home`
6. Click on `Customize your store` task 
7. Click on `Assembler Hub` button
8. Click on `Done` button
9. A transitional page is shown
10. Open the dev console and turn on the "Preserve log" option
11. Click on "Preview store"
12. Observe that `wcadmin_customize_your_store_transitional_preview_store_click` event is recorded.
13. Click on “Go to the Editor”
14.  Observe that `wcadmin_customize_your_store_transitional_editor_click` event is recorded
15. Click on "Back to Home"
16. Observe that `wcadmin_customize_your_store_transitional_home_click` event is recorded

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add track events to customize store transitional page

#### Comment 

</details>

### CODE REVIEW

**Comment by psealock** [comment]
Upon updating localStorage, I should start seeing events logged to the console. Is there a way to enable development mode on Jurassic Ninja? 
```
localStorage.setItem( 'debug', 'wc-admin:*' );
```
https://github.com/woocommerce/woocommerce/blob/36c644a1c4af908b5c1b2060c215b5643dd07648/packages/js/tracks/src/utils.js#L1

**Comment by chihsuan** [comment]
> Upon updating localStorage, I should start seeing events logged to the console.

Hey @psealock Yes, you should see the events after updating load storage and reloading the page.

> Is there a way to enable development mode on Jurassic Ninja?

I'm not sure if it's possible to enable development mode. 🤔 But in production mode, you should be able to see logs. It only affects whether logs are actually sent to the server.

File: plugins/woocommerce-admin/client/customize-store/transitional/index.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+						"Your store is a reflection of your unique style and personality, and we're thrilled to see it come to life.",
+						'woocommerce'
+					) }
+				</h2>
+				<Button
+					className="woocommerce-customize-store__transitional-preview-button"
+					variant="primary"
+					onClick={ () => {
+						recordEvent(
+							'customize_your_transitional_preview_store_click'
```
Comment (adrianduffell) [inline plugins/woocommerce-admin/client/customize-store/transitional/index.tsx]:
```suggestion
							'customize_your_store_transitional_preview_store_click'
```

File: plugins/woocommerce-admin/client/customize-store/transitional/index.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+							{ __(
+								'Head to the Editor to change your images and text, add more pages, and make any further customizations.',
+								'woocommerce'
+							) }
+						</p>
+						<Button
+							variant="tertiary"
+							onClick={ () => {
+								recordEvent(
+									'customize_your_transitional_editor_click'
```
Comment (adrianduffell) [inline plugins/woocommerce-admin/client/customize-store/transitional/index.tsx]:
```suggestion
									'customize_your_store_transitional_editor_click'
```

File: plugins/woocommerce-admin/client/customize-store/transitional/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+							{ __(
+								'Go back to the Home screen to complete your store setup and start selling',
+								'woocommerce'
+							) }
+						</p>
+						<Button
+							variant="tertiary"
+							onClick={ () => {
+								recordEvent(
+									'customize_your_transitional_home_click'
```
Comment (adrianduffell) [inline plugins/woocommerce-admin/client/customize-store/transitional/index.tsx]:
```suggestion
									'customize_your_store_transitional_home_click'
```

**Review by adrianduffell — APPROVED** [review]
Hey @chihsuan , this is all tested great but I think there's just a typo in all the event names.

Pre-approving! 🚀

**Review by rjchow — APPROVED** [review]
The added tracks work fine! 🚀 
I wonder if we should have a track for the transitional page view too, as the 'preview store' button doesn't lead out so we can't just sum up all the tracks for a 100%

**Comment by chihsuan** [comment]
> I wonder if we should have a track for the transitional page view too, as the 'preview store' button doesn't lead out so we can't just sum up all the tracks for a 100%

Good point! I just set up the navigation for the transition page. It fires `wcadmin_page_view` track now when the transition page is loaded.


Screenshot 2023-09-18 at 17 01 46

**Pull request merged**